web_programming/instagram_crawler.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+from __future__ import annotations
+
+import json
+
+import requests
+from bs4 import BeautifulSoup
+
+headers = {
+    "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) "
+    "AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36"
+}
+
+
+def extract_user_profile(script) -> dict:
+    """
+    May raise json.decoder.JSONDecodeError
+    """
+    data = script.contents[0]
+    info = json.loads(data[data.find('{"config"') : -1])
+    return info["entry_data"]["ProfilePage"][0]["graphql"]["user"]
+
+
+class InstagramUser:
+    """
+    Class Instagram crawl instagram user information
+
+    Usage:
+    >>> instagram_user = InstagramUser("github")
+    >>> instagram_user.is_verified
+    True
+    >>> instagram_user.biography
+    'Built for developers.'
+    """
+
+    def __init__(self, username):
+        self.url = f"https://www.instagram.com/{username}/"
+        self.user_data = self.get_json()
+
+    def get_json(self) -> dict:
+        """
+        Return a dict of user information
+        """
+        html = requests.get(self.url, headers=headers).text
+        scripts = BeautifulSoup(html, "html.parser").find_all("script")
+        try:
+            return extract_user_profile(scripts[4])
+        except (json.decoder.JSONDecodeError, KeyError):
+            return extract_user_profile(scripts[3])
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self.username})"
```

Comment:
```suggestion
    def __repr__(self) -> str:
        return f"{self.__class__.__name__}('{self.username}')"
```
Here, the username is going to be a `str` object. `repr` is the string representation that will evaluate to a Python object. A good test to keep in mind while writing the code for `repr` is `obj == eval(repr(obj))`